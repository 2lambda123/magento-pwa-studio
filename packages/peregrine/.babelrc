{
  "plugins": [
    "transform-object-rest-spread",
    "transform-class-properties",
    ["transform-react-jsx", { "pragma": "createElement" }],
    ["transform-runtime", {
      "helpers": true,
      "polyfill": false, // polyfills will be handled by preset-env
      "regenerator": false
    }]
  ],
  "presets": [
    ["env", {
      "targets": {
        "browsers": ["> 5%"]
      },
      "modules": false
    }]
  ],
  "env": {
    "test": {
      "plugins": [
        "transform-es2015-modules-commonjs",
        "transform-object-rest-spread",
        "transform-class-properties",
        ["transform-react-jsx", { "pragma": "createElement" }]
      ]
    }
  }
}
