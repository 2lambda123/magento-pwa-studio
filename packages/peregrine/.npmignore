.eslintrc.js
.storybook
/lib/**/__{docs,helpers,mocks,stories,tests}__/**
/scripts
