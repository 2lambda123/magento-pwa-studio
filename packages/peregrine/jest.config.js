module.exports = {
    displayName: 'Peregrine',
    setupFiles: [
        '<rootDir>/scripts/shim.js',
        '<rootDir>/scripts/fetch-mock.js'
    ],
    browser: true,
    testURL: 'https://localhost/',
    coveragePathIgnorePatterns: ['scripts/*', 'node_modules', 'src/index.js'],
    testPathIgnorePatterns: ['dist', 'node_modules']
};
