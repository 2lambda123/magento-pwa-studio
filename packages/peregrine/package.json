{
  "name": "@magento/peregrine",
  "version": "3.0.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "clean": " ",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist"
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@storybook/react": "~4.1.13",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "10.x",
    "yarn": ">=1.12.0"
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
