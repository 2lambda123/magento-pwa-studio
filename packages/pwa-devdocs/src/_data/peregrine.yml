title: Peregrine
entries:
  - label: Overview
    url: /peregrine/

  - label: Storybook UI testing
    url: /peregrine/testing/storybook/

  - label: Utility modules
    entries:

      - label: REST API client
        url: /peregrine/reference/rest-api-client/

  - label: React components
    entries:

      - label: ContainerChild
        url: /peregrine/reference/container-child/

      - label: Item
        url: /peregrine/reference/item/

      - label: Items
        url: /peregrine/reference/items/

      - label: List
        url: /peregrine/reference/list/

      - label: Router
        url: /peregrine/reference/router/
