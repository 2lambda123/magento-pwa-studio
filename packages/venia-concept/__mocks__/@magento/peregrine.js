import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/src');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useDocumentListener = jest.fn(peregrine.useDocumentListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);

// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useDocumentListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam
};
