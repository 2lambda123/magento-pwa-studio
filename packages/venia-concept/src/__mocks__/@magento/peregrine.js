import { createElement } from 'react';

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
export const Price = () => <div />;
