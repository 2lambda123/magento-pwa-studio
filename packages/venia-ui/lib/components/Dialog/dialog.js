import React from 'react';
import { bool, func, shape, string, object } from 'prop-types';
import { Form } from 'informed';
import { X as CloseIcon } from 'react-feather';

import { useScrollLock } from '@magento/peregrine';
import { mergeClasses } from '@magento/venia-ui/lib/classify';

import Button from '../Button';
import Icon from '../Icon';
import { Modal } from '../Modal';
import defaultClasses from './dialog.css';

/**
 * The Dialog component shows its children content in a dialog,
 * encoding the look-and-feel and behavior in one place for consistency across the app.
 *
 * @typedef Dialog
 * @kind functional component
 *
 * @param {Object}  props
 * @param {Object}  props.classes - A set of class overrides to apply to elements.
 * @param {String}  props.cancelText - The text to display on the Dialog cancel button.
 * @param {String}  props.confirmText - The text to display on the Dialog confirm button.
 * @param {Object}  props.formProps - Props to apply to the internal form. @see https://joepuzzo.github.io/informed/?path=/story/form--props.
 * @param {Boolean} props.isModal - Determines behavior of clicking on the mask. False cancels Dialog.
 * @param {Boolean} props.isOpen - Whether the Dialog is currently showing.
 * @param {Func}    props.onCancel - A function to call when the user cancels the Dialog.
 * @param {Func}    props.onConfirm - A function to call when the user confirms the Dialog.
 * @param {Boolean} props.shouldDisableAllButtons - A toggle for whether the buttons should be disabled.
 * @param {Boolean} props.shouldDisableConfirmButton - A toggle for whether the confirm button should be disabled.
 *                                                     The final value is OR'ed with shouldDisableAllButtons.
 * @param {String}  props.title - The title of the Dialog.
 */
const Dialog = props => {
    const {
        cancelText,
        children,
        confirmText,
        formProps,
        isModal,
        isOpen,
        onCancel,
        onConfirm,
        shouldDisableAllButtons,
        shouldDisableConfirmButton,
        title
    } = props;

    // Prevent the page from scrolling in the background
    // when the Dialog is open.
    useScrollLock(isOpen);

    const classes = mergeClasses(defaultClasses, props.classes);
    const rootClass = isOpen ? classes.root_open : classes.root;
    const isMaskDisabled = shouldDisableAllButtons || isModal;
    const confirmButtonDisabled =
        shouldDisableAllButtons || shouldDisableConfirmButton;

    const confirmButtonClasses = {
        root_highPriority: classes.confirmButton
    };

    const maybeCloseXButton = !isModal ? (
        <button
            className={classes.headerButton}
            disabled={shouldDisableAllButtons}
            onClick={onCancel}
            type="reset"
        >
            <Icon src={CloseIcon} />
        </button>
    ) : null;

    return (
        <Modal>
            <aside className={rootClass}>
                <Form
                    className={classes.form}
                    {...formProps}
                    onSubmit={onConfirm}
                >
                    {/* The Mask. */}
                    <button
                        className={classes.mask}
                        disabled={isMaskDisabled}
                        onClick={onCancel}
                        type="reset"
                    />

                    {/* The Dialog. */}
                    <div className={classes.dialog}>
                        <div className={classes.header}>
                            <span className={classes.headerText}>{title}</span>
                            {maybeCloseXButton}
                        </div>
                        <div className={classes.body}>
                            <div className={classes.contents}>{children}</div>
                            <div className={classes.buttons}>
                                <Button
                                    disabled={shouldDisableAllButtons}
                                    onClick={onCancel}
                                    priority="normal"
                                    type="reset"
                                >
                                    {cancelText}
                                </Button>
                                <Button
                                    classes={confirmButtonClasses}
                                    disabled={confirmButtonDisabled}
                                    priority="high"
                                    type="submit"
                                >
                                    {confirmText}
                                </Button>
                            </div>
                        </div>
                    </div>
                </Form>
            </aside>
        </Modal>
    );
};

export default Dialog;

Dialog.propTypes = {
    cancelText: string,
    classes: shape({
        body: string,
        cancelButton: string,
        confirmButton: string,
        container: string,
        contents: string,
        header: string,
        headerText: string,
        headerButton: string,
        mask: string,
        root: string,
        root_open: string
    }),
    confirmText: string,
    formProps: object,
    isModal: bool,
    isOpen: bool,
    onCancel: func,
    onConfirm: func,
    shouldDisableAllButtons: bool,
    shouldDisableSubmitButton: bool,
    title: string
};

Dialog.defaultProps = {
    cancelText: 'Cancel',
    confirmText: 'Confirm',
    isModal: false
};
