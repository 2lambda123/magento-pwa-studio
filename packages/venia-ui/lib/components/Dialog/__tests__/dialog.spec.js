import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import Dialog from '../dialog';

jest.mock('../../../classify');
jest.mock('../../Modal', () => ({
    Modal: props => <modal-mock>{props.children}</modal-mock>
}));

const props = {
    isOpen: true,
    shouldDisableAllButtons: false,
    title: 'Unit Test Dialog Title'
};

test('renders a basic Dialog', () => {
    // Act.
    const wrapper = createTestInstance(<Dialog {...props} />);

    // Assert.
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('renders a Dialog with disabled buttons', () => {
    // Arrange.
    const myProps = {
        ...props,
        shouldDisableAllButtons: true
    };

    // Act.
    const wrapper = createTestInstance(<Dialog {...myProps} />);

    // Assert.
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('renders a dialog with only the confirm button disabled', () => {
    // Arrange.
    const myProps = {
        ...props,
        shouldDisableConfirmButton: true
    };

    // Act.
    const wrapper = createTestInstance(<Dialog {...myProps} />);

    // Assert.
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('supports modifying title and button texts', () => {
    // Arrange.
    const myProps = {
        ...props,
        cancelText: 'Unit Test Cancel',
        confirmText: 'Unit Test Confirm',
        title: 'Unit Test Title'
    };

    // Act.
    const wrapper = createTestInstance(<Dialog {...myProps} />);

    // Assert.
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('does not render a close X button in modal mode', () => {
    // Arrange.
    const myProps = {
        ...props,
        isModal: true
    };

    // Act.
    const wrapper = createTestInstance(<Dialog {...myProps} />);

    // Assert.
    expect(wrapper.toJSON()).toMatchSnapshot();
});
