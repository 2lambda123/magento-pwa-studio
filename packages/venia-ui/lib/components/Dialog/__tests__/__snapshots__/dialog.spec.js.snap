// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render a close X button in modal mode 1`] = `
<modal-mock>
  <aside>
    <form
      onKeyDown={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <button
        disabled={true}
        type="reset"
      />
      <div>
        <div>
          <span>
            Unit Test Dialog Title
          </span>
        </div>
        <div>
          <div />
          <div>
            <button
              className="root_normalPriority"
              disabled={false}
              type="reset"
            >
              <span
                className="content"
              >
                Cancel
              </span>
            </button>
            <button
              className="root_highPriority"
              type="submit"
            >
              <span
                className="content"
              >
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </aside>
</modal-mock>
`;

exports[`renders a Dialog with disabled buttons 1`] = `
<modal-mock>
  <aside>
    <form
      onKeyDown={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <button
        disabled={true}
        type="reset"
      />
      <div>
        <div>
          <span>
            Unit Test Dialog Title
          </span>
          <button
            disabled={true}
            type="reset"
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
        </div>
        <div>
          <div />
          <div>
            <button
              className="root_normalPriority"
              disabled={true}
              type="reset"
            >
              <span
                className="content"
              >
                Cancel
              </span>
            </button>
            <button
              className="root_highPriority"
              disabled={true}
              type="submit"
            >
              <span
                className="content"
              >
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </aside>
</modal-mock>
`;

exports[`renders a basic Dialog 1`] = `
<modal-mock>
  <aside>
    <form
      onKeyDown={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <button
        disabled={false}
        type="reset"
      />
      <div>
        <div>
          <span>
            Unit Test Dialog Title
          </span>
          <button
            disabled={false}
            type="reset"
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
        </div>
        <div>
          <div />
          <div>
            <button
              className="root_normalPriority"
              disabled={false}
              type="reset"
            >
              <span
                className="content"
              >
                Cancel
              </span>
            </button>
            <button
              className="root_highPriority"
              type="submit"
            >
              <span
                className="content"
              >
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </aside>
</modal-mock>
`;

exports[`renders a dialog with only the confirm button disabled 1`] = `
<modal-mock>
  <aside>
    <form
      onKeyDown={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <button
        disabled={false}
        type="reset"
      />
      <div>
        <div>
          <span>
            Unit Test Dialog Title
          </span>
          <button
            disabled={false}
            type="reset"
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
        </div>
        <div>
          <div />
          <div>
            <button
              className="root_normalPriority"
              disabled={false}
              type="reset"
            >
              <span
                className="content"
              >
                Cancel
              </span>
            </button>
            <button
              className="root_highPriority"
              disabled={true}
              type="submit"
            >
              <span
                className="content"
              >
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </aside>
</modal-mock>
`;

exports[`supports modifying title and button texts 1`] = `
<modal-mock>
  <aside>
    <form
      onKeyDown={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <button
        disabled={false}
        type="reset"
      />
      <div>
        <div>
          <span>
            Unit Test Title
          </span>
          <button
            disabled={false}
            type="reset"
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
        </div>
        <div>
          <div />
          <div>
            <button
              className="root_normalPriority"
              disabled={false}
              type="reset"
            >
              <span
                className="content"
              >
                Unit Test Cancel
              </span>
            </button>
            <button
              className="root_highPriority"
              type="submit"
            >
              <span
                className="content"
              >
                Unit Test Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </aside>
</modal-mock>
`;
